src/github.com/matrix-org/dendrite/roomserver/storage/previous_events_table.go

```diff
@@ -0,0 +1,72 @@
+package storage
+
+import (
+	"database/sql"
+	"github.com/matrix-org/dendrite/roomserver/types"
+)
+
+const previousEventSchema = `
+-- The previous events table stores the event_ids referenced by the events
+-- stored in the events table.
+-- This is used to tell if a new event is already referenced by an event we 
+CREATE TABLE IF NOT EXISTS previous_events (
+	// The string event ID taken from the prev_events key of an event.
+	previous_event_id TEXT NOT NULL,
+	// The SHA256 reference hash taken from the prev_events key of an event.
+	previous_reference_sha256 BYTEA NOT NULL,
+	// A list of numeric event IDs of events that reference this prev_event.
+	// This isn'
```

Comment:
Presumably this isn't guaranteed to contain unique entries given the `array_append` you do on `:32`.